### PR TITLE
Add main function and tests

### PR DESCRIPTION
**Pull request by fchorney**
Adding the `main.jl` file, and tests for it.

### CODE REVIEW

File: src/utilities/new_versions.jl [hunk 1]
```diff
@@ -162,7 +164,8 @@ end
 
 function continue_with_pr(dep::DepInfo, bump_compat_containing_equality_specifier::Bool)
     # Determine if we need to make a new PR
-    if !isnothing(dep.version_spec) && dep.latest_version in dep.version_spec
+    if (!isnothing(dep.version_spec) && !isnothing(dep.latest_version)) &&
+       dep.latest_version in dep.version_spec
```
Comment (fchorney) [inline src/utilities/new_versions.jl:168]: looks like this change is required because `nothing in VersionSpec` doesn't have a method for it.

**Comment by fchorney** [comment]
bors try

**Review by mattBrzezinski — APPROVED** [review]
LGTM! After this I suppose we test it, then add in the GitLab configurability stuff.

**Comment by mattBrzezinski** [comment]
bors try

**Pull request merged**